Source/ASDisplayNode.mm

```diff
@@ -433,6 +467,7 @@ - (void)onDidLoad:(ASDisplayNodeDidLoadBlock)body
 - (void)dealloc
 {
   _flags.isDeallocating = YES;
+  [self baseWillDealloc];
```

Comment:
Short answer: so that any calls that come out of `baseWillDealloc` will know not to retain self.

Medium length answer: The method is _base_ will dealloc, indicating the base class is going to run its dealloc routine. The flag refers to the _object_ deallocation which in fact began before this method was called, so we try to keep up with it.

Long answer: Technically flags like these are unsafe, even though they're common. The only correct way to find out if an object is deallocating (unsafe to retain) is to try and retain a weak pointer to the object. As of right now, if a subclass called something from its `dealloc`, our flag wouldn't be set and we might try to retain self. The safe thing to do is to register a weak pointer to `self` during `init` and replace this flag, with an attempt to retain that.